### PR TITLE
che #13813: Pushing rhel images to 'quay.io/openshiftio' and centos images to 'quay.io/eclipse-che' during the CI build

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?
Pushing rhel images to 'quay.io/openshiftio' and centos images to 'quay.io/eclipse-che' during the CI build

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13813

Should be merged only once dedicated cico job pr is merged - https://github.com/openshiftio/openshiftio-cico-jobs/pull/1035

### CODE REVIEW

**Comment by ibuziuk** [comment]
The only thing I'm not sure about - are we already pushing *rhel* (not centos) images to `quay.io/openshiftio` - https://quay.io/repository/openshiftio/che-devfile-registry?tab=tags ?
the image name is `che-devfile-registry` and I would expect that rhel image name would be `rhel-che-devfile-registry`

File: cico_build.sh [hunk 1]
```diff
@@ -69,17 +71,31 @@ function deploy() {
...
-  # Let's deploy
+  # Let's push 'rhel' images to 'quay.io/openshiftio'
   docker build -t ${IMAGE} -f ${DOCKERFILE} .
 
   TAG=$(echo "$GIT_COMMIT" | cut -c1-"${DEVSHIFT_TAG_LEN}")
 
   tag_push "${REGISTRY}/openshiftio/$IMAGE:$TAG"
   tag_push "${REGISTRY}/openshiftio/$IMAGE:latest"
-  echo 'CICO: Image pushed, ready to update deployed app'
+  echo "CICO: Images pushed to 'quay.io/openshiftio', ready to update deployed app"
```
Comment (l0rd) [inline cico_build.sh]: A few suggestions:
- I don't think you should login/push to `quay.io/eclipse-che` if the `"$TARGET" == "rhel" `
- You don't need to the `docker build` twice
- I would avoid tagging the images as `latest` because that should be done when we do the release of Che every 3 weeks (especially for eclipse-che org)

Comment (l0rd) [reply cico_build.sh]: And do we really need to push centos images to both openshiftio and eclipse-che?

Comment (ibuziuk) [reply cico_build.sh]: > I don't think you should login/push to quay.io/eclipse-che if the "$TARGET" == "rhel"
> You don't need to the docker build twice

Maybe I'm missing smth but in the issue description there is - `Update che-devfile-registry-master CentOS CI job to push rhel images to quay.io/openshiftio and centos images to quay.io/eclipse-che` and my understanding is that master CI job always build 'rhel for now, right?. So, I do not really understand how we can avoid building and pushing images twice if we want CentOS images beeing build as part of the same job

Comment (ibuziuk) [reply cico_build.sh]: it is actually related to my question in my first question in the PR - https://github.com/eclipse/che-devfile-registry/pull/31#issuecomment-510394593

Comment (l0rd) [reply cico_build.sh]: If what you are saying was true ($TARGET is always "rhel") the script would build `rhel` images **only** (the script does only one docker build). But we know that this is not the case because we see that centos images are pushed to the registry right? 

So I think that this script is called twice: once with not specific $TARGET and once with TARGET=rhel. And we should double check with SD team but I suspect that we (you and I) do not have the privileges to see the rhel images even if they are pushed in that org.

Comment (l0rd) [reply cico_build.sh]: Another thing I would add in this PR is an update to the README file to use the `eclipse-che` org images instead of the openshiftio.

Comment (ibuziuk) [reply cico_build.sh]: > So I think that this script is called twice: once with not specific $TARGET and once with TARGET=rhel

@l0rd looks like this is the case (script is called twice with & without the target), because:

- for now there is only one job for devfile registry - https://github.com/openshiftio/openshiftio-cico-jobs/blob/master/devtools-ci-index.yaml#L4470
- centos images are pushed to https://quay.io/repository/openshiftio/che-devfile-registry?tab=tags
- on dsaas devfile-registry is build against `openshiftio/rhel-che-devfile-registry` images (and yes, we do not have access to https://quay.io/repository/openshiftio/rhel-che-devfile-registry)

**Comment by ibuziuk** [comment]
@l0rd PR has been updated: 
- during the build rhel images would be pushed to `quay.io/openshiftio/rhel-che-devfile-registry`
- centos images will be pushed to `quay.io/eclipse-che/rhel-che-devfile-registry`
- https://quay.io/repository/openshiftio/che-devfile-registry should be deprecated

I plan to add yet another commit with docs & build instructions but before want to clarify your point around `latest` tag.
Basically, all the instructions in Readme are currently depended on this tag: e.g.:

- https://github.com/eclipse/che-devfile-registry/blob/master/build.sh#L11
- https://github.com/eclipse/che-devfile-registry/blob/master/deploy/openshift/che-devfile-registry.yaml#L89

if brand-new `quay.io/eclipse-che/rhel-che-devfile-registry` will not contain the latest tag, which one should be used ?

**Comment by benoitf** [comment]
> centos images will be pushed to quay.io/eclipse-che/rhel-che-devfile-registry

why name is rhel-something if it's centos ?

**Comment by ibuziuk** [comment]
@benoitf sorry, typo - updated

**Comment by benoitf** [comment]
just a small note, on dockerhub we had `eclipse/che-something` as org was eclipse.
but on quay.io now with `eclipse-che` org, do we still `che` in image name ?

`eclipse-che/che-devfile-registry` --> `eclipse-che/devfile-registry` ?

**Comment by ibuziuk** [comment]
@benoitf `che-devfile-registry` is the repo name so, I personally would expect to have the very same name for docker image, but I do understand your concern

**Comment by benoitf** [comment]
@ibuziuk then it would mean quay org should the same than github org or that we move to eclipse-che org on github as well (it's possible to have its own 'eclipse' org so it could be a good idea as well so we cleary see only repositories linked to che)

**Comment by ibuziuk** [comment]
@benoitf agree, https://quay.io/organization/eclipse would be more logical
@l0rd wdyt ?

**Comment by l0rd** [comment]
I think eclipse organisation makes more sense as well. But didn't knew if we had the time to move to that org. 

For the change to `latest` I am proposing to:

1. stop tagging every build as `latest` (without actually changing the README etc...) => in this PR
2. start tagging nightlies with the new cron job and update the README etc...to use nigthlies
3. start tagging releases every 3 weeks as latest

But if you want to stay on the safe side (meaning that we want to make sure that the README will always point to the latest build) you should swap 1. and 2. and do that in separate PRs.

**Comment by l0rd** [comment]
Anyway LGTM for the changes you have made so far

File: cico_build.sh (hunk 1 above)
Comment (rhopp) [reply cico_build.sh]: Yep.. The script is run twice. Once here:
https://github.com/openshiftio/openshiftio-cico-jobs/blob/master/devtools-ci-index.yaml#L891
And for the second time here (where `extra_target=rhel`):
https://github.com/openshiftio/openshiftio-cico-jobs/blob/master/devtools-ci-index.yaml#L902

**Comment by ibuziuk** [comment]
> I think eclipse organisation makes more sense as well. But didn't knew if we had the time to move to that org.

Indeed looks like we do not have time for moving from one org to anoter and setup ci bots for those

>stop tagging every build as latest (without actually changing the README etc...) => in this PR

Done

@l0rd sounds good, so no docs update in this PR. Could you please approve if you are ok to merge it in the current state.

P.S.

> So I think that this script is called twice: once with not specific $TARGET and once with TARGET=rhe

checked with SD and indeed `ci_cmd` command is executed twice (for `rhel` and `centos` builds) - https://github.com/openshiftio/openshiftio-cico-jobs/blob/master/devtools-ci-index.yaml#L880-L909

**Comment by ibuziuk** [comment]
https://quay.io/repository/eclipse-che/che-devfile-registry?tab=tags \o/
prod-preview has been also updated successfully \o/
